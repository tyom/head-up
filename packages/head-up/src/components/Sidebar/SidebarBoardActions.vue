<template>
  <div class="SidebarBoardActions">
    <SidebarButton
      class="add-button"
      title="Shortcut: a"
      @click="$emit('add-board')"
    >
      <v-icon name="plus"/>
    </SidebarButton>
    <SidebarButton
      :class="{_active: editMode}"
      class="edit-button"
      title="Shortcut: e"
      @click="$emit('toggle-edit')"
    >
      <v-icon name="edit"/>
    </SidebarButton>
  </div>
</template>

<script>
import SidebarButton from './SidebarButton';

export default {
  components: {
    SidebarButton,
  },
  props: {
    editMode: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
.SidebarBoardActions {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 0.25em;
}
</style>
