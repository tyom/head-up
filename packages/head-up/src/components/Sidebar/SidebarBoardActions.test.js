import { shallowMount } from '@vue/test-utils';
import SidebarBoardActions from './SidebarBoardActions';

const provideMock = {
  isEditing: () => false,
};

test('render default', () => {
  const wrapper = shallowMount(SidebarBoardActions, {
    provide: provideMock,
  });
  expect(wrapper).toMatchSnapshot();
});

test('render edit-toggled', () => {
  const wrapper = shallowMount(SidebarBoardActions, {
    provide: {
      isEditing: () => true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('respond to actions', () => {
  const wrapper = shallowMount(SidebarBoardActions, {
    provide: provideMock,
  });

  wrapper.find('.add-button').vm.$emit('click');
  wrapper.find('.edit-button').vm.$emit('click');

  expect(wrapper.emitted('add-board')[0]).toBeTruthy();
  expect(wrapper.emitted('toggle-edit')[0]).toBeTruthy();
});
