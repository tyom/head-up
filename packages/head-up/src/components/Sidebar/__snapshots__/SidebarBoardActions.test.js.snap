// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default 1`] = `
<div class="SidebarBoardActions">
  <sidebarbutton-stub title="Shortcut: a" class="add-button">
    <v-icon-stub name="plus"></v-icon-stub>
  </sidebarbutton-stub>
  <sidebarbutton-stub title="Shortcut: e" class="edit-button">
    <v-icon-stub name="edit"></v-icon-stub>
  </sidebarbutton-stub>
</div>
`;

exports[`render edit-toggled 1`] = `
<div class="SidebarBoardActions">
  <sidebarbutton-stub title="Shortcut: a" class="add-button">
    <v-icon-stub name="plus"></v-icon-stub>
  </sidebarbutton-stub>
  <sidebarbutton-stub title="Shortcut: e" class="edit-button _active">
    <v-icon-stub name="edit"></v-icon-stub>
  </sidebarbutton-stub>
</div>
`;
